gtest/src/wasm_executor.rs

```diff
@@ -37,58 +42,80 @@ impl WasmExecutor {
+        let module =
+            Module::new(&engine, program.code().code()).map_err(|error| error.to_string())?;
 
         let mut memory =
             WasmtimeMemory::new(&mut store, MemoryType::new(program.static_pages().0, None))
-                .expect("Failed to create memory");
+                .map_err(|error| error.to_string())?;
+
         let funcs = funcs_tree::build(&mut store, memory, None);
         let mut externs = Vec::with_capacity(module.imports().len());
         for import in module.imports() {
             if import.module() != "env" {
-                panic!("Non environment import in module");
+                return Err(format!(
+                    "Non environment import in module: {:?}",
+                    import.module()
+                ));
             }
             match import.name() {
                 Some("memory") => externs.push(Extern::Memory(memory)),
                 Some(key) => {
                     if funcs.contains_key(key) {
                         externs.push(funcs[key].into())
                     } else {
-                        panic!("Wasm is asking for unknown function: {:?}. Consider to add in from FuncsHandler", key)
+                        return Err(format!("Wasm is asking for function: {:?} with wasn't found. Consider to add in from FuncsHandler", key));
                     }
                 }
                 _ => continue,
             };
         }
 
         let instance =
-            Instance::new(&mut store, &module, &externs).expect("Failed to create instance");
-        WasmExecutor::set_pages(&mut store, &mut memory, memory_pages)
-            .expect("Failed to set memory pages");
+            Instance::new(&mut store, &module, &externs).map_err(|error| error.to_string())?;
+        WasmExecutor::set_pages(&mut store, &mut memory, memory_pages)?;
 
-        Self {
+        Ok(Self {
             instance,
             store,
             memory,
-        }
+        })
     }
 
     /// Executes non-void function by provided name.
-    /// Panics if no function with such name was found or function was void
-    pub(crate) fn execute(&mut self, function_name: &str) -> Vec<u8> {
-        let function = self.get_function(function_name);
+    /// Panics if function is void
+    pub(crate) fn execute(&mut self, function_name: &str) -> Result<Vec<u8>, String> {
+        let function = self.get_function(function_name)?;
         let mut prt_to_result_array = [Val::I32(0)];
 
         function
             .call(&mut self.store, &[], &mut prt_to_result_array)
-            .expect("Failed call");
+            .map_err(|error| self.get_error_message(error))?;
 
         match prt_to_result_array[0] {
             Val::I32(ptr_to_result) => self.read_result(ptr_to_result),
-            _ => panic!("{}", "Got wrong type"),
+            _ => Err("Got wrong type".to_string()),
+        }
+    }
+
+    fn get_error_message(&self, error: Error) -> String {
```

Comment:
As far as I remember all our errors implement the `Display` trait so this function looks excessive.

Reply:
Removed this function.